microsetta_private_api/repo/tests/test_perk_fulfillment_repo.py

```diff
@@ -0,0 +1,538 @@
+            test_daklapack_order_result
+    ):
+        verify_address_result.return_value = VERIFY_ADDRESS_DICT
+        test_daklapack_order_result.return_value = {
+            "order_address": "wedontcareaboutthis",
+            "order_success": True,
+            "order_id": DUMMY_ORDER_ID
+        }
+
+        with Transaction() as t:
+            # create a dummy Daklapack order
+            acct_repo = AccountRepo(t)
+            submitter_acct = acct_repo.get_account(SUBMITTER_ID)
+
+            creation_timestamp = dateutil.parser.isoparse(
+                "2020-10-09T22:43:52.219328Z")
+            last_polling_timestamp = dateutil.parser.isoparse(
+                "2020-10-19T12:40:19.219328Z")
+            desc = "a description"
+            planned_send_date = datetime.date(2032, 2, 9)
+            last_status = "accepted"
+
+            # create dummy daklapack order object
+            input = DaklapackOrder(DUMMY_ORDER_ID, submitter_acct,
+                                   PROJECT_IDS, DUMMY_DAKLAPACK_ORDER, desc,
+                                   planned_send_date, creation_timestamp,
+                                   last_polling_timestamp, last_status)
+
+            # call create_daklapack_order
+            admin_repo = AdminRepo(t)
+            returned_id = admin_repo.create_daklapack_order(input)
+
+            ut = UserTransaction(t)
+            ut.add_transaction(TRANSACTION_ONE_KIT)
+            pfr = PerkFulfillmentRepo(t)
+            res = pfr.process_pending_fulfillments()
+
+            cur = t.cursor()
+
+            # Confirm that the order populated into fundrazr_daklapack_orders
+            cur.execute(
+                "SELECT COUNT(*) "
+                "FROM campaign.fundrazr_daklapack_orders "
+                "WHERE dak_order_id = %s",
+                (returned_id, )
+            )
+            res = cur.fetchone()
+            self.assertEqual(res[0], 1)
+
+    @patch(
+        "microsetta_private_api.repo.perk_fulfillment_repo."
+        "create_daklapack_order_internal"
+    )
+    @patch("microsetta_private_api.repo.interested_user_repo.verify_address")
+    def test_process_pending_fulfillments_one_kit_fail(
+            self,
+            verify_address_result,
+            test_daklapack_order_result
+    ):
+        verify_address_result.return_value = VERIFY_ADDRESS_DICT
+        test_daklapack_order_result.return_value = {
+            "order_address": "wedontcareaboutthis",
+            "order_success": False,
+            "daklapack_api_error_msg": "Some error message",
+            "daklapack_api_error_code": "Some error code"
+        }
+
+        with Transaction() as t:
+            ut = UserTransaction(t)
+            ut.add_transaction(TRANSACTION_ONE_KIT)
+            pfr = PerkFulfillmentRepo(t)
+            res = pfr.process_pending_fulfillments()
+
+            # res is a list of errors, which should be 1
+            self.assertNotEqual(len(res), 0)
```

Comment:
Based on the comment, I think the following change makes sense

```suggestion
            self.assertEqual(len(res), 1)
```